evap/student/views.py

```diff
@@ -111,31 +111,26 @@ def vote(request, course_id):
     # all forms are valid, begin vote operation
     with transaction.atomic():
         # add user to course.voters
-        # not using course.voters.add(request.user) since it fails silently when done twice.
-        # manually inserting like this gives us the 'created' return value and ensures at the database level that nobody votes twice.
-        __, created = course.voters.through.objects.get_or_create(userprofile_id=request.user.pk, course_id=course.pk)
-        if not created:  # vote already got recorded, bail out
-            raise SuspiciousOperation("A second vote has been received shortly after the first one.")
+        # not using course.voters.add(request.user) since that fails silently when done twice.
+        course.voters.through.objects.create(userprofile_id=request.user.pk, course_id=course.pk)
```

Comment:
why don't we have to check for a second vote anymore?

Reply:
because create throws an exception when such an object already exists. using get_or_create and then manually throwing seemed unnecessary to me.

Reply:
👍